src/native/managed/cdacreader/src/Target.cs

```diff
@@ -283,6 +341,70 @@ public bool TryReadGlobalPointer(string name, out TargetPointer pointer)
         return true;
     }
 
+    public TypeInfo GetTypeInfo(DataType type)
+    {
+        if (!TryGetTypeInfo(type, out TypeInfo typeInfo))
+            throw new InvalidOperationException($"Failed to get type info for '{type}'");
+
+        return typeInfo;
+    }
+
+    public bool TryGetTypeInfo(DataType type, out TypeInfo typeInfo)
+        => _knownTypes.TryGetValue(type, out typeInfo);
+
+    public TypeInfo GetTypeInfo(string type)
+    {
+        if (!TryGetTypeInfo(type, out TypeInfo typeInfo))
+            throw new InvalidOperationException($"Failed to get type info for '{type}'");
+
+        return typeInfo;
+    }
```

Comment:
I'm always concerned when I see these split APIs. I don't fully understand why a newly designed system needs both a Try and a Throwing API. We should be able to architect something that is either uses exceptions or doesn't. Mixing the two seems like a code smell in a new code base.

Reply:
that reminds me.  Do we need to wrap everything in Legacy SOSDaclImpl in a `try`/`catch`?

Reply:
My preference would be to always return an `int`, just for consistency, and wrap any COM API implementation with a `try`/`catch`. It permits us to use all .NET APIs and keeps us honest about the boundary.